x-pack/plugins/ml/public/application/explorer/anomaly_timeline.tsx

```diff
@@ -145,26 +163,98 @@ export const AnomalyTimeline: FC<AnomalyTimelineProps> = React.memo(
       [severityUpdate, swimLaneSeverity]
     );
 
+    const openCasesModal = useCallback(
+      (swimLaneType: SwimlaneType) => {
+        selectCaseModal!.open({
+          attachments: [
+            {
+              // @ts-ignore
+              type: 'persistableState' as const,
+              persistableStateAttachmentTypeId: ANOMALY_SWIMLANE_EMBEDDABLE_TYPE,
+              persistableStateAttachmentState: {
+                swimlaneType: swimLaneType,
+                ...(swimLaneType === SWIMLANE_TYPE.VIEW_BY
+                  ? { viewBy: viewBySwimlaneFieldName }
+                  : {}),
+                jobIds: selectedJobs?.map((v) => v.id),
+                timeRange: globalTimeRange,
+              } as AnomalySwimlaneEmbeddableCustomInput,
+              owner: PLUGIN_ID,
+            },
+          ],
+        });
+      },
+      [selectCaseModal, selectedJobs, globalTimeRange, viewBySwimlaneFieldName]
+    );
+
     const annotations = useMemo(() => overallAnnotations.annotationsData, [overallAnnotations]);
 
-    const menuItems = useMemo(() => {
-      const items = [];
+    const menuPanels = useMemo<EuiContextMenuPanelDescriptor[]>(() => {
+      const rootItems = [] as EuiContextMenuPanelItemDescriptor[];
+      const panels = [{ id: 0, items: rootItems }] as EuiContextMenuPanelDescriptor[];
+
       if (canEditDashboards) {
-        items.push(
-          <EuiContextMenuItem
-            key="addToDashboard"
-            onClick={setIsAddDashboardActive.bind(null, true)}
-            data-test-subj="mlAnomalyTimelinePanelAddToDashboardButton"
-          >
+        rootItems.push({
+          name: (
             <FormattedMessage
               id="xpack.ml.explorer.addToDashboardLabel"
               defaultMessage="Add to dashboard"
             />
-          </EuiContextMenuItem>
-        );
+          ),
+          onClick: setIsAddDashboardActive.bind(null, true),
+          'data-test-subj': 'mlAnomalyTimelinePanelAddToDashboardButton',
+        });
       }
-      return items;
-    }, [canEditDashboards]);
+
+      const canCreateCase = true;
```

Comment:
Are you going to add in user privilege checking in this PR, or do it in a follow-up. Currently the action is showing up for a user who has `Read` permission for Cases in all three solutions, which leads to an error when trying to attach the swim lane:

<img width="610" alt="image" src="https://user-images.githubusercontent.com/7405507/185900735-7c91f981-b104-44d2-b35d-cf6685019cff.png">

Reply:
@jonathan-buttner do we have a bug in this 'Select case' modal here? I have set up a role which only has `All` permissions for Cases for Security and Stack. In the modal the solution dropdown correctly only shows Security and Stack (cases), but I can still see the Observability cases in the table:

<img width="778" alt="image" src="https://user-images.githubusercontent.com/7405507/185902281-a818f307-49a3-4762-8d00-8fb06ab02856.png">

Reply:
> Are you going to add in user privilege checking in this PR, or do it in a follow-up? Currently the action is showing up for a user who has `Read` permission for Cases in all three solutions, which leads to an error when trying to attach the swim lane:
> 
> <img alt="image" width="610" src="https://user-images.githubusercontent.com/7405507/185900735-7c91f981-b104-44d2-b35d-cf6685019cff.png">

@peteharverson added a check in https://github.com/elastic/kibana/pull/138994/commits/f27ada82e707beb9421bdd87a46f4cd4a3d0e10e

Reply:
@peteharverson yikes, could we hop on a zoom so I could see? The backend should be preventing that user from viewing the observability cases.

Reply:
Sorry, just to clarify, this occurs when the user has `read` for a solution - the solution dropdown doesn't contain the solution which has `read` for cases, but the case still appears in the table, allowing the user to select it, at which point you see the following error:

<img width="462" alt="image" src="https://user-images.githubusercontent.com/7405507/185930513-f6f58d95-0c6e-43db-a65f-31dc13a3af43.png">

We need to be checking that the user has `all` access so that they can add a comment to the case with the ML embeddable. I'll create a separate issue @jonathan-buttner

Reply:
FYI - created https://github.com/elastic/kibana/issues/139206 for the issue with the 'select case' modal.